### PR TITLE
[tests] skip running expected fail tests

### PR DESCRIPTION
**Pull request by bukepo**
This commit skip running expected fail tests so that it will be easier
to find real failures and saves some time.

### CODE REVIEW

File: tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py [hunk 1]
```diff
@@ -43,6 +43,8 @@
 
 
 class Cert_5_3_10_AddressQuery(thread_cert.TestCase):
+    support_ncp = False
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py:46]:
```suggestion
    SUPPORT_NCP = False
```

File: tests/scripts/thread-cert/thread_cert.py [hunk 2]
```diff
@@ -50,7 +51,21 @@
...
+    """
+
+    support_ncp = True
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if os.getenv('NODE_TYPE', 'sim') == 'ncp-sim' and not self.support_ncp:
+            # 77 means skip this test case in automake tests
+            sys.exit(77)
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/thread_cert.py:65]: nit: move `super().__init__(*args, **kwargs)` below `sys.exit` ?

File: tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py (hunk 1 above)
Comment (bukepo) [reply tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py:46]: this will be done in a separate PR together with the topology attribute.

Comment (simonlingoogle) [reply tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py:46]: fine to apply style changes in a separate PR

**Pull request merged**